js/views/message_detail_view.js

```diff
@@ -98,14 +132,21 @@
             }));
             this.view.$el.prependTo(this.$('.message-container'));
 
+            this.grouped = _.groupBy(this.model.get('errors'), 'number');
             if (this.model.isOutgoing()) {
-                this.conversation.contactCollection.reject(function(c) {
+                var contacts = this.conversation.contactCollection.reject(function(c) {
                     return c.isMe();
-                }).forEach(this.renderContact.bind(this));
+                });
+
+                _.sortBy(contacts, function(c) {
+                    var prefix = this.grouped[c.id] ? '0' : '1';
```

Comment:
Adding a prefix to the title strings gives us both alphabetical sorting and sorting by errors. It will be like this:

```
[
'0Alice', // with errors
'0Bob', // with errors
'1Alice (GVoice)',
'1John'
]
```